core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -256,11 +280,15 @@ protected void parse() throws ProtocolException {
         length = cursor - offset;
     }
 
-    public static Block createGenesis(NetworkParameters n) {
-        Block genesisBlock = new Block(n, BLOCK_VERSION_GENESIS);
+    public static Block createGenesis(NetworkParameters n, long time, long difficultyTarget, long nonce) {
+        Transaction t = createGenesisTransaction(n, genesisTxInputScriptBytes, FIFTY_COINS, genesisTxScriptPubKeyBytes);
+        return new Block(n, BLOCK_VERSION_GENESIS, time, difficultyTarget, nonce, Collections.singletonList(t));
+    }
+
+    @VisibleForTesting
+    public static Block createGenesisUnitTest(NetworkParameters n) {
         Transaction t = createGenesisTransaction(n, genesisTxInputScriptBytes, FIFTY_COINS, genesisTxScriptPubKeyBytes);
-        genesisBlock.addTransaction(t);
-        return genesisBlock;
+        return new Block(n, BLOCK_VERSION_GENESIS, Utils.currentTimeSeconds(), Block.EASIEST_DIFFICULTY_TARGET, Collections.singletonList(t));
```

Comment:
After this refactoring (I don't want to change the behavior in the refactoring) we should probably try replacing `Utils.currentTimeSeconds()` with a call to directly get (un-mockable) system time.